### PR TITLE
string_utils has FILETIME_to_string_UTC. New UT project. New INT proje…

### PR DESCRIPTION
**Pull request by anporumb**
…ct. New specs

### CODE REVIEW

File: interfaces/devdoc/string_utils.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+`string_utils` requirements
+============
```
Comment (mattdurak) [inline interfaces/devdoc/string_utils.md:2]: Should do the `#` titles instead of `====`

File: interfaces/devdoc/string_utils.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## Overview
+
+`string_utils` is a collection of string utils. The set of requirements in this document is far from complete.
```
Comment (mattdurak) [inline interfaces/devdoc/string_utils.md:6]: >collection of string utils [](http://example.com/codeflow?start=20&length=26)

consider a short list of what utilities we have right now, like:
- printf helpers
- wide/narrow conversions
- FILETIME to string helpers

File: interfaces/devdoc/string_utils.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+MOCKABLE_INTERFACE(string_utils_printf,
+    /*produces a string as if printed by vprintf*/
+    FUNCTION(, char*, vsprintf_char, const char*, format, va_list, va),
+    /*produces a string as if printed by vwprintf*/
+    FUNCTION(, wchar_t*, vsprintf_wchar, const wchar_t*, format, va_list, va)
+)
+
+/*takes a FILETIME, returns a nice string representation of it*/
+MOCKABLE_FUNCTION(, char*, FILETIME_toAsciiArray, const FILETIME*, fileTime);
```
Comment (mattdurak) [inline interfaces/devdoc/string_utils.md:29]: >toAsciiArray [](http://example.com/codeflow?start=36&length=12)

an interesting name...should probably rename this and the new one so it is a little more clear what they do and what the difference is.

File: interfaces/devdoc/string_utils.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    FUNCTION(, char*, wcs_to_mbs, const wchar_t*, source)
+)
+```
+
+### FILETIME_to_string_UTC
+```c
+MOCKABLE_FUNCTION(, char*, FILETIME_to_string_UTC, const FILETIME*, fileTime);
+```
+
+`FILETIME_to_string_UTC` returns a human readable representation of `fileTime`. Here's an example: `(SYSTEMTIME){ /* 2022-06-09T18:59:48.476Z */ .wYear = 2022, .wMonth = 06, .wDayOfWeek = 4, .wDay = 09, .wHour = 18, .wMinute = 59, .wSecond = 48, .wMilliseconds = 476 }`. The returned string needs to be `free`d by the caller.
```
Comment (mattdurak) [inline interfaces/devdoc/string_utils.md:48]: > human readable representation [](http://example.com/codeflow?start=34&length=30)

According to your comment in the other file, it's also a C initializer format, which is maybe interesting to call out here.

**Review by mattdurak — APPROVED** [review]
:shipit:

File: interfaces/devdoc/string_utils.md (hunk 1 above)
Comment (anporumb) [reply interfaces/devdoc/string_utils.md]: done here and everywhere in this repo

File: interfaces/devdoc/string_utils.md (hunk 2 above)
Comment (anporumb) [reply interfaces/devdoc/string_utils.md]: added

File: interfaces/devdoc/string_utils.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (anporumb) [reply interfaces/devdoc/string_utils.md]: Removed "human", removed C-style. 

[https://msazure.visualstudio.com/One/_workitems/edit/14684336](https://msazure.visualstudio.com/One/_workitems/edit/14684336)

File: interfaces/devdoc/string_utils.md (hunk 3 above)
Comment (anporumb) [reply interfaces/devdoc/string_utils.md:28]: addressed by [https://msazure.visualstudio.com/One/_workitems/edit/14684336](https://msazure.visualstudio.com/One/_workitems/edit/14684336)

File: interfaces/devdoc/string_utils.md (hunk 4 above)
Comment (anporumb) [reply interfaces/devdoc/string_utils.md]: Removed the designated initializer part.

**Pull request merged**